contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,964 @@
+            /**
+             * It can happen that a withdrawal (or a full withdrawal) transactions would
+             * remove all of the liquidity from the token with a NFT token still existing.
+             * In that case the token can not be staked into the gauge, as some liquidity
+             * needs to be added to it first.
+             */
+            if (_getLiquidity() > 0) {
+                positionManager.approve(address(clGauge), tokenId);
+                clGauge.deposit(tokenId);
+            }
+        }
+    }
+
+    // slither-disable-end reentrancy-no-eth
+
+    /// @notice the constructor
+    /// @param _stratConfig the basic strategy configuration
+    /// @param _wethAddress Address of the Erc20 WETH Token contract
+    /// @param _oethbAddress Address of the Erc20 OETHb Token contract
+    /// @param _swapRouter Address of the Aerodrome Universal Swap Router
+    /// @param _nonfungiblePositionManager Address of position manager to add/remove
+    ///         the liquidity
+    /// @param _clPool Address of the Aerodrome concentrated liquidity pool
+    /// @param _clGauge Address of the Aerodrome slipstream pool gauge
+    /// @param _sugarHelper Address of the Aerodrome Sugar helper contract
+    /// @param _lowerBoundingTick Smaller bounding tick of our liquidity position
+    /// @param _upperBoundingTick Larger bounding tick of our liquidity position
+    /// @param _tickClosestToParity Tick that is closer to 1:1 price parity
+    constructor(
+        BaseStrategyConfig memory _stratConfig,
+        address _wethAddress,
+        address _oethbAddress,
+        address _swapRouter,
+        address _nonfungiblePositionManager,
+        address _clPool,
+        address _clGauge,
+        address _sugarHelper,
+        int24 _lowerBoundingTick,
+        int24 _upperBoundingTick,
+        int24 _tickClosestToParity
+    ) InitializableAbstractStrategy(_stratConfig) {
+        require(
+            _lowerBoundingTick == _tickClosestToParity ||
+                _upperBoundingTick == _tickClosestToParity,
+            "Miss configured tickClosestToParity"
+        );
+
+        WETH = _wethAddress;
+        OETHb = _oethbAddress;
+        swapRouter = ISwapRouter(_swapRouter);
+        positionManager = INonfungiblePositionManager(
+            _nonfungiblePositionManager
+        );
+        clPool = ICLPool(_clPool);
+        clGauge = ICLGauge(_clGauge);
+        helper = ISugarHelper(_sugarHelper);
+        sqrtRatioX96TickLower = ISugarHelper(_sugarHelper).getSqrtRatioAtTick(
+            _lowerBoundingTick
+        );
+        sqrtRatioX96TickHigher = ISugarHelper(_sugarHelper).getSqrtRatioAtTick(
+            _upperBoundingTick
+        );
+        sqrtRatioX96TickClosestToParity = ISugarHelper(_sugarHelper)
+            .getSqrtRatioAtTick(_tickClosestToParity);
+
+        lowerTick = _lowerBoundingTick;
+        upperTick = _upperBoundingTick;
+        tickSpacing = 1;
```

Comment:
This is always hardcoded to `1`, why not make it a constant?

Reply:
Actually IDK why we didn't read it from the pool. Fixed here: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/0ec09164e6cdef621a54031f1866027858ee8d46